flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/hint/FlinkHints.java

```diff
@@ -119,7 +120,8 @@ public static Optional<String> getTableAlias(RelNode node) {
     public static boolean canTransposeToTableScan(RelNode node) {
         return node instanceof LogicalProject // computed column on table
                 || node instanceof LogicalFilter
-                || node instanceof LogicalSnapshot;
+                || node instanceof LogicalSnapshot
+                || node instanceof WatermarkAssigner;
```

Comment:
From reading through the code a little bit, I'm trying to figure out if there could be any issues with how `FlinkHints.canTransposeToTableScan` is called in `QueryHintsResolver.getTableScan`. 

I'm guessing that a `WatermarkAssigner` never makes it to that method -- if it does, seems like this would change the behavior a little bit.

Reply:
In fact it was one of the root causes
`WatermarkAssigner` came here however in case of `WatermarkAssigner` the method returned false and then empty result for hinted tables and finally as a result tables from hints were not matched in that case